core/camera/src/main/java/com/google/jetpackcamera/core/camera/effects/processors/LowLightBoostSurfaceProcessor.kt

```diff
@@ -0,0 +1,198 @@
+        inputSurfaceRequestScope = newScope
+
+        outputSurfaceFlow.value?.surfaceOutput?.let { outputSurface ->
+            coroutineScope.launch {
+                val surface = outputSurface.getSurface(glExecutor, {})
+                tryCreateLowLightBoostSession(surface)
+            }
+        }
+
+        // Clean up when the request is released (e.g., camera is closed)
+        request.addRequestCancellationListener(glExecutor) { // Replace context if not available
+            Log.d(TAG, "InputSurfaceRequest cancelled: $request")
+            inputSurfaceRequestScope?.cancel("SurfaceRequest cancelled by CameraX.")
+            inputSurfaceRequestScope = null
+            releaseLowLightBoostSession()
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.R)
+    private suspend fun tryCreateLowLightBoostSession(outputSurfaceForLlb: Surface) {
+        inputSurfaceRequestScope?.withSurfaceRequest { currentInputRequest ->
+
+            coroutineScope.launch { // Launch in your processor's scope
+                try {
+                    // Create LowLightBoostOptions with the output surface for LLB
+                    // and dimensions from the input SurfaceRequest.
+                    val llbOptions = LowLightBoostOptions(
+                        outputSurfaceForLlb, // This is where LLB writes its output
+                        cameraId,
+                        currentInputRequest.resolution.width,
+                        currentInputRequest.resolution.height,
+                        true
+                    )
+
+                    if (lowLightBoostSession != null) {
+                        releaseLowLightBoostSession()
+                    }
+
+                    lowLightBoostSession = lowLightBoostClient
+                        .createSession(llbOptions, createLowLightBoostCallback())
+                        .await()
+
+                    sessionContainer.lowLightBoostSession = lowLightBoostSession
+
+                    // Get the input surface from the LowLightBoostSession
+                    val llbInputSurface = lowLightBoostSession?.getCameraSurface()
+                        ?: throw IllegalStateException(
+                            "LowLightBoostSession did not provide an input surface."
+                        )
+
+                    Log.d(TAG, "LLB Session created. Providing LLB input surface to CameraX.")
+
+                    // Fulfill the CameraX SurfaceRequest with LLB's input surface
+                    currentInputRequest.provideSurface(llbInputSurface, glExecutor) { result ->
+                        Log.d(TAG, "CameraX SurfaceRequest result: ${result.resultCode}")
+                        llbInputSurface.release()
+                        when (result.resultCode) {
+                            SurfaceRequest.Result.RESULT_SURFACE_USED_SUCCESSFULLY -> {
+                                Log.i(TAG, "CameraX is using LLB input surface.")
```

Comment:
The result is sent _after_ CameraX is done using the surface. I believe we should be releasing the LLB session in every case here, no matter the result.

Reply:
I think in that case, release is called in the onSessionDisconnected callback. Until version beta06 is available, calling release multiple times will result in an error being thrown. I tried adding `releaseLowLightBoostSession()` to both of the cases where it's not currently being called, and in both circumstances, JCA would crash.